master/internal/rm/kubernetesrm/kubernetes_resource_manager.go

```diff
@@ -63,20 +71,320 @@ func New(
+	if k.config.DefaultComputeResourcePool == "" {
+		return sproto.GetDefaultComputeResourcePoolResponse{}, rmerrors.ErrNoDefaultResourcePool
+	}
+	return sproto.GetDefaultComputeResourcePoolResponse{PoolName: k.config.DefaultComputeResourcePool}, nil
+}
+
+// GetExternalJobs implements rm.ResourceManager.
+func (ResourceManager) GetExternalJobs(sproto.GetExternalJobs) ([]*jobv1.Job, error) {
+	return nil, rmerrors.ErrNotSupported
+}
+
+// GetJobQ implements rm.ResourceManager.
+func (k *ResourceManager) GetJobQ(msg sproto.GetJobQ) (map[model.JobID]*sproto.RMJobInfo, error) {
+	if msg.ResourcePool == "" {
+		msg.ResourcePool = k.config.DefaultComputeResourcePool
+	}
+
+	rp, err := k.poolByName(msg.ResourcePool)
+	if err != nil {
+		return nil, err
+	}
+	resp := rp.GetJobQ(msg)
+	return resp, nil
+}
+
+// GetJobQueueStatsRequest implements rm.ResourceManager.
+func (k *ResourceManager) GetJobQueueStatsRequest(
+	*apiv1.GetJobQueueStatsRequest,
+) (*apiv1.GetJobQueueStatsResponse, error) {
+	resp := &apiv1.GetJobQueueStatsResponse{
+		Results: make([]*apiv1.RPQueueStat, 0),
+	}
+
+	for poolName, rp := range k.pools {
+		qStats := apiv1.RPQueueStat{
+			ResourcePool: poolName,
+			Stats:        rp.GetJobQStats(sproto.GetJobQStats{}),
+		}
+
+		aggregates, err := k.fetchAvgQueuedTime(poolName)
+		if err != nil {
+			return nil, fmt.Errorf("fetch average queued time: %s", err)
+		}
+		qStats.Aggregates = aggregates
+
+		resp.Results = append(resp.Results, &qStats)
+	}
+
+	return resp, nil
+}
+
+// GetResourcePools implements rm.ResourceManager.
+func (k *ResourceManager) GetResourcePools(*apiv1.GetResourcePoolsRequest) (*apiv1.GetResourcePoolsResponse, error) {
+	summaries := make([]*resourcepoolv1.ResourcePool, 0, len(k.poolsConfig))
+	for _, pool := range k.poolsConfig {
+		summary, err := k.createResourcePoolSummary(pool.PoolName)
+		if err != nil {
+			// Should only raise an error if the resource pool doesn't exist and that can't happen.
+			// But best to handle it anyway in case the implementation changes in the future.
+			return nil, err
+		}
+
+		jobStats, err := k.getPoolJobStats(pool)
+		if err != nil {
+			return nil, err
+		}
+
+		summary.Stats = jobStats
+		summaries = append(summaries, summary)
+	}
+	return &apiv1.GetResourcePoolsResponse{ResourcePools: summaries}, nil
+}
+
+// GetSlot implements rm.ResourceManager.
+// TODO(!!!): Implement GetSlot for Kubernetes RM.
+func (ResourceManager) GetSlot(*apiv1.GetSlotRequest) (*apiv1.GetSlotResponse, error) {
+	return nil, rmerrors.ErrNotSupported
+}
+
+// GetSlots implements rm.ResourceManager.
+// TODO(!!!): Implement GetSlots for Kubernetes RM.
+func (ResourceManager) GetSlots(*apiv1.GetSlotsRequest) (*apiv1.GetSlotsResponse, error) {
+	return nil, rmerrors.ErrNotSupported
+}
+
+// MoveJob implements rm.ResourceManager.
+// TODO(!!!): This should know which pool it wants.
+func (k *ResourceManager) MoveJob(msg sproto.MoveJob) error {
```

Comment:
will these TODOs be addressed in this PR?

Reply:
i'll ticket them before i merge and update the todos